src/hooks/users.ts

```diff
@@ -105,6 +105,31 @@ export const useRecentWorldUsers = (): {
   );
 };
 
+export const useRecentWorldUser = (
+  userId: string
+): {
+  user?: WithId<User>;
+  userLastSeenIn?: string;
+} => {
+  const { recentWorldUsers } = useRecentWorldUsers();
+
+  const user = useMemo(() => {
+    return recentWorldUsers.find((worldUser) => worldUser.id === userId);
+  }, [userId, recentWorldUsers]);
+
+  const userLastSeenIn = Object.keys(user?.lastSeenIn ?? {})[0];
+
+  return { user, userLastSeenIn };
+};
+
+export const useRecentUserStatus = (userId: string): RecentUserStatusType => {
+  const { user } = useRecentWorldUser(userId);
+
+  if (user?.status) return RecentUserStatusType.busy;
+
+  return user ? RecentUserStatusType.online : RecentUserStatusType.offline;
+};
```

Comment:
this should be implemented as a util function. That function will accept `user` obj. `getUserDisplayStatus`

Reply:
https://github.com/sparkletown/sparkle/pull/1484/commits/0e4e679fca9153d9b74759272963562f5f6903fc